### PR TITLE
:sparkles: Add Connections.StartConnection outputs

### PR DESCRIPTION
**Pull request by rmtuckerphx**
## Proposed Changes

Add the following Output classes for Connections.StartConnection:

- connection://AMAZON.VerifyPerson/2
- connection://AMAZON.LinkApp/2
- connection://AMAZON.PrintPDF/1
- connection://AMAZON.PrintImage/1
- connection://AMAZON.PrintWebPage/1
- connection://AMAZON.TestStatusCode/1
- connection://AMAZON.AskForPermissionsConsent/2
- connection://AMAZON.ScheduleTaxiReservation/1
- connection://AMAZON.ScheduleFoodEstablishmentReservation/1

## Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project
- [x] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

### CODE REVIEW

File: platforms/platform-alexa/src/output/templates/ConnectionAskForPermissionConsentOutput.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { BaseOutput, Output, OutputOptions, OutputTemplate, EnumLike } from '@jovotech/framework';
+
+export enum ConsentLevel {
+  Account = 'ACCOUNT',
+  Person = 'PERSON',
+}
+
+export type ConsentLevelLike = EnumLike<ConsentLevel> | string;
+
+export enum PermissionScope {
```
Comment (aswetlow) [inline platforms/platform-alexa/src/output/templates/ConnectionAskForPermissionConsentOutput.ts]: Should we move this to `platform-alexa/src/interfaces.ts` ? This part is also used in `platform-alexa/src/output/models/card/Card.ts`

**Comment by aswetlow** [comment]
Thank you for the pr @rmtuckerphx

File: platforms/platform-alexa/src/output/templates/ConnectionAskForPermissionConsentOutput.ts (hunk 1 above)
Comment (rmtuckerphx) [reply platforms/platform-alexa/src/output/templates/ConnectionAskForPermissionConsentOutput.ts]: @aswetlow 
I refactored some interfaces, types and enums to `interfaces.ts`.
I reused the `PermissionScope` in `Card.ts` while allowing Card to have a few extra values.

**Comment by rmtuckerphx** [comment]
@aswetlow I had to do another reactor to models/common to support other card classes.
There is an issue with the code that I don't know enough TypeScript to fix. The `PermissionScope` values supported in the Connections delegate are a subset of those supported by Cards.

I tried to handle it this way:

```
export enum CardPermissionScope {
  ReadList = 'read::alexa:household:list',
  WriteList = 'write::alexa:household:list',
  ReadAddressFull = 'read::alexa:device:all:address',
}

export type CardPermissionScopeLike = EnumLike<PermissionScope> | EnumLike<CardPermissionScope> | string;
```

But `AskForRemindersPermissionOutput` doesn't like `CardPermissionScopeLike.ReadWriteReminders`

**Comment by aswetlow** [comment]
Thank you for the changes.

Splitting `PermissionScope` into `PermissionScope` and `CardPermissionScope` would make it a breaking change.

Is it necessary to have `CardPermissionScope` ?

**Comment by rmtuckerphx** [comment]
@aswetlow There are 3 more PermissionScope values that are supported in cards than are supported by Connections.

Our options are:
1. Breaking change where CardPermissionScope  has the 3 additional items
2. Use a single list of PermissionScope  values for both Cards and Connection even though 3 would be invalid for Connections.
3. Keep PermissionScope the same and have a different ConnectionPermissionScope that include duplicates from both lists.

Originally I created it closer to option 3. The current PR is option 1.

Tell me what option you want me to implement.

**Comment by aswetlow** [comment]
Thank you for the explanation. It makes sense now. We discuss it internally and I'll come back to you.

**Comment by jankoenig** [comment]
Thank you @rmtuckerphx! I've just added some docs for this.

Since I haven't worked with Skill Connections myself, I'm not sure if I missed something, could you take another look?

Also, according to the [Alexa docs](https://developer.amazon.com/en-US/docs/alexa/custom-skills/use-skill-connections-to-request-tasks.html#for-managed-skill-connection-with-resume_session), there is usually an `onCompletion` property that I couldn't find in the output classes I was looking at. Any thoughts on this?

Thanks!

**Comment by rmtuckerphx** [comment]
@jankoenig The docs look good. 
Good catch on `onCompletion`
@aswetlow  I've added `onCompletion` to each `ConnectionXxxOutput` class.

**Review by jankoenig — APPROVED** [review]
Thank you @rmtuckerphx! Ready to merge from my side ✅ 

@aswetlow and I are going to talk about the interface issues tomorrow

**Comment by aswetlow** [comment]
> @aswetlow There are 3 more PermissionScope values that are supported in cards than are supported by Connections.
> 
> Our options are:
> 
> 1. Breaking change where CardPermissionScope  has the 3 additional items
> 2. Use a single list of PermissionScope  values for both Cards and Connection even though 3 would be invalid for Connections.
> 3. Keep PermissionScope the same and have a different ConnectionPermissionScope that include duplicates from both lists.
> 
> Originally I created it closer to option 3. The current PR is option 1.
> 
> Tell me what option you want me to implement.

Hey @rmtuckerphx 
Sorry for the late response. We discussed it and came to the conclusion that 3. is the best option. 

In my opinion, duplicates are not always bad because they often increase comprehensibility.

**Comment by jankoenig** [comment]
> Hey @rmtuckerphx Sorry for the late response. We discussed it and came to the conclusion that 3. is the best option.
> 
> In my opinion, duplicates are not always bad because they often increase comprehensibility.

I just made the according changes and fixed merge conflicts.

Ready to merge from my side ✅

**Review by aswetlow — APPROVED** [review]

**Comment by jankoenig** [comment]
Thank you @rmtuckerphx 🎉

**Pull request merged**